### DISCUSSION TITLE
MQTT client ID

### DISCUSSION THREAD

**Discussion by borti-16** [tbnobody/OpenDTU#1854] [Q&A]
Hi There,
maybe a dumb question... does openDTU send a client ID along? If so is this a fixed value or is there any chance to set a specific ID? I tried to find an answer in the code but failed.

**Comment by datenheim** [comment]
Not a dumb question IMO.
Currently trying to connect to a Thingsboard instance, where the client ID is required to establish a connection.
But there is no setting anywhere in the UI in openDTU.
So I thought it might use the default client ID from the Espressif Core which would be:
> [client_id](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/protocols/mqtt.html#_CPPv4N24esp_mqtt_client_config_t13credentials_t9client_idE): pointer to the client ID, defaults to ESP32_%CHIPID% where %CHIPID% are the last 3 bytes of MAC address in hex format
But this is still not the solution right now.
Question remains: which client ID is really used by openDTU?
